From 51a58592e96fd1efe446329b4b55e36025e49015 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Tue, 14 Sep 2021 13:54:03 +0300
Subject: [PATCH] [SW-55646] gaudi2: arc: move <>_get_img_addr

Move lkd_dev_spec_get_img_addr from arc common to
specific architecture of onix,asic, and pldm.

Change-Id: I40b15bbb17ae3cd3c4465e82c4f67e8844b82037
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index f942998..9101c29 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -16,6 +16,17 @@
 #include "common.h"
 #include <hl.h>
 
+#define SDRAM_BASE_FULL 0x1000FFFFFD000000ull
+
+u64 lkd_dev_spec_get_img_addr(uintptr_t base)
+{
+	/* SPSRAM fit can only be in general region */
+	if ((base >= SP_SRAM_BASE) && (base < SP_SRAM_BASE + HL_SP_SRAM_SIZE))
+		return (base & ~SP_SRAM_BASE) | HL_AXI_GENERAL_BASE;
+
+	return (base & ~SDRAM_BASE) | SDRAM_BASE_FULL;
+}
+
 u32 hl_skip_pcie_init(void)
 {
 	/* TODO: Use DumpSel[*] to skip PCIe initialization */
diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 67cec60..1c2f93f 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -43,20 +43,6 @@
 #define mmPCIE_WRAP_VUART_TX_DATA       0x4C15F60
 #define mmPCIE_WRAP_VUART_TX_STATUS     0x4C15F64
 
-#define ARC_REL_ADDR_MASK	0x000000000FFFFFFFULL
-#define HL_AXI_GENERAL_BASE	0x1000007FF0000000ULL
-
-u64 lkd_dev_spec_get_img_addr(uintptr_t base)
-{
-	uintptr_t base_offset = base & ARC_REL_ADDR_MASK; /* offset in general region */
-
-	/* SPSRAM fit can only be in general region */
-	if ((base >= SP_SRAM_BASE) && (base < SP_SRAM_BASE + HL_SP_SRAM_SIZE))
-		return base_offset | HL_AXI_GENERAL_BASE;
-
-	return base_offset | SDRAM_BASE_FULL;
-}
-
 void preboot_putc_virt_uart(const char c)
 {
 	/*
diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 5a35dd5..1471b3a 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -14,6 +14,7 @@
 #define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
 
+#define HL_AXI_GENERAL_BASE	0x1000007FF0000000ULL
 #define HL_SP_SRAM_BASE		0x1000007FF7FE0000ULL
 #define HL_SP_SRAM_SIZE		0x10000
 #define SDRAM_BASE		0x10000000
diff --git a/arch/gaudi2/arc/onix.c b/arch/gaudi2/arc/onix.c
index 4643ccb..64914e9 100644
--- a/arch/gaudi2/arc/onix.c
+++ b/arch/gaudi2/arc/onix.c
@@ -11,9 +11,15 @@
 
 #include "common.h"
 
-
 #define EFUSE_SYSCFG_ONIX_DEFAULT	0
 
+#define SDRAM_BASE_FULL	0x1001000200000000ull
+
+u64 lkd_dev_spec_get_img_addr(uintptr_t base)
+{
+	return (base & ~SDRAM_BASE) | SDRAM_BASE_FULL;
+}
+
 u32 hl_spi_baudr_get(void)
 {
 	u32 factor = 2; /* Min Divider */
diff --git a/arch/gaudi2/arc/pldm.c b/arch/gaudi2/arc/pldm.c
index a6987cc..55a0b00 100644
--- a/arch/gaudi2/arc/pldm.c
+++ b/arch/gaudi2/arc/pldm.c
@@ -8,6 +8,12 @@
 
 #include <preboot.h>
 
+#define SDRAM_BASE_FULL 0x1000FFFFFD000000ull
+
+u64 lkd_dev_spec_get_img_addr(uintptr_t base)
+{
+	return (base & ~SDRAM_BASE) | SDRAM_BASE_FULL;
+}
 
 u32 hl_skip_pcie_init(void)
 {
diff --git a/arch/gaudi2/common/common.h b/arch/gaudi2/common/common.h
index 76a6ffc..5423566 100644
--- a/arch/gaudi2/common/common.h
+++ b/arch/gaudi2/common/common.h
@@ -23,7 +23,6 @@
 	#define DECODER_ENABLED_MASK 0x1AB
 	#define NIC_PORTS_MASK 0x0
 	#define TPC_ENABLED_MASK 0x1020101
-	#define SDRAM_BASE_FULL 0x1000FFFFFD000000ull
 #elif TARGET_IS_ONIX
 	#define MME_MASK 0
 	#define EDMA_ENABLED_MASK 0x0
@@ -31,7 +30,6 @@
 	#define DECODER_ENABLED_MASK 0x1AB
 	#define NIC_PORTS_MASK 0
 	#define TPC_ENABLED_MASK 0x1020101
-	#define SDRAM_BASE_FULL	0x1001000200000000ull
 #else /* ASIC */
 	#define MME_MASK 0xF
 	#define EDMA_ENABLED_MASK 0xFF
@@ -39,7 +37,6 @@
 	#define DECODER_ENABLED_MASK 0x1FF
 	#define NIC_PORTS_MASK 0xFFFFFF
 	#define TPC_ENABLED_MASK 0x1FFFFFF
-	#define SDRAM_BASE_FULL 0x1000FFFFFD000000ull
 #endif
 
 #define NUM_OF_DCORES			4
